en/docs/how-tos/analytics/elk-based-analytics.md

```diff
@@ -0,0 +1,362 @@
+    | "errorCode"                  | int       | The error code generated in a fault.                                 |
+    | "errorMessage"               | string    | The error message associated with the fault.                         |
+    | "errorType"                  | string    | The type of error (e.g., THROTTLED).                                 |
+
+### Step 2 - Configuring ELK
+
+#### Installing Elasticsearch
+
+1. [Install Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/8.1/install-elasticsearch.html) according to your operating system.
+
+2. Make sure Elasticsearch is [up and running](https://www.elastic.co/guide/en/elasticsearch/reference/8.1/configuring-stack-security.html).
+
+!!! info
+    As recommended by ELK, a minimum 3 node cluster is required for a production environment.
+
+#### Installing Filebeat
+
+1. [Install Filebeat](https://www.elastic.co/guide/en/beats/filebeat/8.1/filebeat-installation-configuration.html#installation) according to your operating system.
+
+2. Configure **Filebeats** to read the log file in the `repository/logs` folder.
+
+    ``` yaml
+    filebeat.inputs:
+    -   type: log
+        enabled: true
+        paths:
+            - {MGW_HOME}/logs/apim_metrics.log
+        include_lines: ['(apimMetrics):']
+    output.logstash:
+    # The Logstash hosts
+         hosts: ["{LOGSTASH_URL}:5044"]
+    ```
+
+#### Installing Logstash
+
+1. [Install Logstash](https://www.elastic.co/guide/en/logstash/8.1/installing-logstash.html) according to your operating system.
+
+    ``` java
+    input {
+        beats {
+            port => 5044
+        }
+    }
+
+    filter {
+        grok {
+            match => ["message", "%{GREEDYDATA:UNWANTED}\ apimMetrics:%{GREEDYDATA:apimMetrics}\, %{GREEDYDATA:UNWANTED} \:%{GREEDYDATA:properties}"]
+        }
+        json {
+            source => "properties"
+        }
+    }
+    output {
+        if [apimMetrics] == " apim:response" {
+            elasticsearch {
+                hosts => ["http://{ELK_URL}:9200"]
+                index => "apim_event_response"
+                user => "elastic"
+                password => "Admin1234"
+            }
+        } else if [apimMetrics] == " apim:faulty" {
+            elasticsearch {
+                hosts => ["http://{ELK_URL}:9200"]
+                index => "apim_event_faulty"
+                user => "elastic"
+                password => "Admin1234"
+            }
+        }
+    }
+    ```
+
+#### Installing Kibana
+
+1. [Install Kibana](https://www.elastic.co/guide/en/kibana/8.1/install.html#_install_kibana_yourself) according to your operating system.
+
+2. [Launch](https://www.elastic.co/guide/en/kibana/8.1/access.html#log-on-to-the-web-application) the Kibana web interface.
+
+3. Log in to the Kibana dashboards.
+
+4. Navigate to Stack Management > index pattern. If you already have any index patterns created under the following names, delete them before importing the saved artifacts.
```

Comment:
```suggestion
4. Navigate to Stack Management > index pattern. If you already have any index pattern created under the following names, delete them before importing the saved artifacts.
```

Reply:
This sentence is continued implying having multiple patterns (Referred to as "delete them" next line). Is there a reason to change this?

Reply:
Done with https://github.com/wso2/docs-mg/pull/184/commits/9971bd66c89b08168724a05788aba122e0886d99